src/lightning/pytorch/plugins/precision/amp.py

```diff
@@ -74,15 +74,19 @@ def optimizer_step(  # type: ignore[override]
         if isinstance(optimizer, LBFGS):
             raise MisconfigurationException("AMP and the LBFGS optimizer are not compatible.")
         closure_result = closure()
+        skipped_backward = closure_result is None
+
+        # If backward was skipped in automatic optimization (return None), unscaling is not needed
+        skip_unscaling = skipped_backward and model.automatic_optimization
 
-        if not _optimizer_handles_unscaling(optimizer):
+        if not _optimizer_handles_unscaling(optimizer) and not skip_unscaling:
             # Unscaling needs to be performed here in case we are going to apply gradient clipping.
             # Optimizers that perform unscaling in their `.step()` method are not supported (e.g., fused Adam).
             # Note: `unscale` happens after the closure is executed, but before the `on_before_optimizer_step` hook.
             self.scaler.unscale_(optimizer)
 
         self._after_closure(model, optimizer)
-        skipped_backward = closure_result is None
+
         # in manual optimization, the closure does not return a value
         if not model.automatic_optimization or not skipped_backward:
```

Comment:
If you apply DeMorgan's law, this is the same

```suggestion
        if not skip_unscaling:
```

and then you could just have `skip_unscaling` and remove `skipped_backward`